pkg/tcl/apitcl/v1/testworkflows.go

```diff
@@ -236,6 +241,83 @@ func (s *apiTCL) PreviewTestWorkflowHandler() fiber.Handler {
 	}
 }
 
+func (s *apiTCL) ExecuteTestWorkflowHandler() fiber.Handler {
+	return func(c *fiber.Ctx) (err error) {
+		name := c.Params("id")
+		errPrefix := fmt.Sprintf("failed to execute test workflow '%s'", name)
+		workflow, err := s.TestWorkflowsClient.Get(name)
+		if err != nil {
+			return s.ClientError(c, errPrefix, err)
+		}
+
+		// Load the execution request
+		var request testkube.TestWorkflowExecutionRequest
+		err = c.BodyParser(&request)
+		if err != nil && !errors.Is(err, fiber.ErrUnprocessableEntity) {
+			return s.BadRequest(c, errPrefix, "invalid body", err)
+		}
+		if request.Name == "" {
+			request.Name = rand.Name()
+		}
+
+		machine := expressionstcl.NewMachine().
+			Register("execution.id", request.Name)
+
+		// Fetch the templates
+		tpls := testworkflowresolver.ListTemplates(workflow)
+		tplsMap := make(map[string]testworkflowsv1.TestWorkflowTemplate, len(tpls))
+		for tplName := range tpls {
+			tpl, err := s.TestWorkflowTemplatesClient.Get(tplName)
+			if err != nil {
+				return s.BadRequest(c, errPrefix, "fetching error", err)
+			}
+			tplsMap[tplName] = *tpl
+		}
+
+		// Apply the configuration
+		_, err = testworkflowresolver.ApplyWorkflowConfig(workflow, mappers2.MapConfigValueAPIToKube(request.Config))
+		if err != nil {
+			return s.BadRequest(c, errPrefix, "configuration", err)
+		}
+
+		// Resolve the TestWorkflow
+		err = testworkflowresolver.ApplyTemplates(workflow, tplsMap)
+		if err != nil {
+			return s.BadRequest(c, errPrefix, "resolving error", err)
+		}
+
+		// Process the TestWorkflow
+		bundle, err := testworkflowprocessor.NewFullFeatured(s.ImageInspector).
+			Bundle(c.Context(), workflow, machine)
+		if err != nil {
+			return s.BadRequest(c, errPrefix, "processing error", err)
+		}
+
+		// Deploy the resources
+		// TODO: rollback on failure
+		for _, item := range bundle.Secrets {
+			_, err = s.Clientset.CoreV1().Secrets(s.Namespace).Create(context.Background(), &item, metav1.CreateOptions{})
+			if err != nil {
+				return s.BadRequest(c, errPrefix, "creating secret", err)
+			}
+		}
+		for _, item := range bundle.ConfigMaps {
+			_, err = s.Clientset.CoreV1().ConfigMaps(s.Namespace).Create(context.Background(), &item, metav1.CreateOptions{})
+			if err != nil {
+				return s.BadRequest(c, errPrefix, "creating configmap", err)
+			}
```

Comment:
do we clear them after job finish?

Reply:
We will, but not at this point. The [**TKC-1651**](https://linear.app/kubeshop/issue/TKC-1651/testworkflows-prepare-controller-for-testworkflows-in-kubernetes) ticket is for building the Controller, which will be handling the Job/Pod/Containers events, logs, and the cleanup 👍

Reply:
great